cmd/dra-example-controller/driver.go

```diff
@@ -94,9 +94,17 @@ func (d driver) GetClaimParameters(ctx context.Context, claim *resourcev1.Resour
 
 func (d driver) Allocate(ctx context.Context, claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) (*resourcev1.AllocationResult, error) {
 	if selectedNode == "" {
-		return nil, fmt.Errorf("TODO: immediate allocations not yet supported")
+		return d.allocateImmediateClaim(claim, claimParameters, class, classParameters)
 	}
 
+	if err := d.allocateDelayedClaim(claim, claimParameters, class, classParameters, selectedNode); err != nil {
+		return nil, err
```

Comment:
We should wrap the error return here with a:
```
fmt.Errorf("error performing delayed allocation: %v", err)
```

Reply:
Done.